### PR TITLE
[TIMOB-25309] Add support for Xcode 9's multiple simulator instances.

### PR DESCRIPTION
**Pull request by cb1kenobi**

### CODE REVIEW

File: lib/simulator.js [hunk 1]
```diff
@@ -985,44 +985,75 @@ function launch(simHandleOrUDID, options, callback) {
...
 
-								if (/^shutdown/i.test(sim.state)) {
-									// the udid that is supposed to be running isn't, kill the simulator
-									emitter.emit('log-debug', __('%s Simulator is running, but udid %s is shut down, stopping simulator', handle.name, handle.udid));
-									stop(handle, next);
-									return;
-								}
+								function waitToBoot() {
+									emitter.emit('log-debug', __('Waiting for sim to boot...'));
+									simctl.waitUntilBooted({ simctl: handle.simctl, udid: handle.udid, timeout: 30000 }, function (err, _booted) {
```
Comment (hansemannn) [inline lib/simulator.js]: Might this be a good time to launch the Simulator earlier like some (..) other platforms do? Basically trigger it as soon as we get the UUID, so we won't have any delay for the initial launch anymore.

**Comment by cb1kenobi** [comment]
All instances of `uuid` should be `udid`. I'll fix this in ioslib@2.0.0 which I'm working on.

File: lib/simulator.js (hunk 1 above)
Comment (cb1kenobi) [reply lib/simulator.js]: We can't launch the simulator and then later install the app. ioslib's API isn't designed to do this. I can fix it, but I'd rather allocate the time towards ioslib v2 where I'll fix all these issues.

**Comment by hansemannn** [comment]
Can we agree on all `Simulator` instead of `sim`? Even if just in the changed files.

**Comment by cb1kenobi** [comment]
`Simulator` is a class name. For variables, `simulator` is too long, but `sim` is perfect. In function and method names, `*Simulator*` is descriptive and perfect.

**Comment by hansemannn** [comment]
Only talking about logs ^^

**Comment by cb1kenobi** [comment]
Oooooooh, that's cool. Is it OK if I tack it onto this PR quick?

**Comment by hansemannn** [comment]
Of course :-D And guess we can merge here if you agree.

**Comment by cb1kenobi** [comment]
@hansemannn Verbiage updated! :)

**Comment by cb1kenobi** [comment]
Merging to get the show on the road.

**Pull request merged**